internal/fs/dir_handle.go

```diff
@@ -242,42 +261,54 @@ func (dh *dirHandle) ensureEntries(ctx context.Context) (err error) {
 //
 // LOCKS_REQUIRED(dh.Mu)
 // LOCKS_EXCLUDED(du.in)
-func (dh *dirHandle) ReadDir(
+func (dh *DirHandle) ReadDir(
 	ctx context.Context,
 	op *fuseops.ReadDirOp) (err error) {
 	// If the request is for offset zero, we assume that either this is the first
 	// call or rewinddir has been called. Reset state.
+
 	if op.Offset == 0 {
-		dh.entries = nil
-		dh.entriesValid = false
-	}
+		dh.Entries = nil
```

Comment:
what if two ls -lh are done in parallel, will all this work with out any issues?

Reply:
Had issues with the previous implementation.Changed it now.